docs/gauge_agg.md

```diff
@@ -0,0 +1,76 @@
+# Gauge Aggregates [<sup><mark>experimental</mark></sup>](/docs/README.md#tag-notes)
+
+A gauge is a metric similar to a counter, with the primary difference being
+that it measures a value that varies up and down over time, rather than an
+ever-increasing COUNT of the number of times something happened.
+Examples include resource utilization metrics, precipitation levels,
+or temperatures.
+
+`gauge_agg` currently shares implementation with `counter_agg` but without the
```

Comment:
This feels inside baseball, which is fine here, but it would be good to figure out some more user facing explanations here at some point.

Reply:
@davidkohn88 I think the expectation is that we're using this section more as developer level docs and as a driver for our doctester, and less as anything user facing.  The goal is to not be having the same material duplicated here and in the on-line docs, but of course, this is very much an evolving design.